### PR TITLE
Merge run_arm.sh & run_amd.sh in one script

### PR DESCRIPTION
**Pull request by msazanov**
Refactoring run script code
Updated dockerfile & dockerfile.armhf
Fix lose data after restart addon
bump version to 1.1.8

### CODE REVIEW

File: zwave_me_server/Dockerfile [hunk 1]
```diff
@@ -6,21 +6,21 @@ ENV distro=focal
...
 RUN apt-get update && \
-    apt-get install -y wget procps gpg libcurl4 logrotate tzdata
+    apt-get install -qqy --no-install-recommends \
+    ca-certificates curl libnss-mdns \
+    wget procps gpg iproute2 openssh-client logrotate
 
-# Add Z-Wave.Me repository
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 5b2f88a91611e683 && \
-    echo "deb https://repo.z-wave.me/z-way/$branch $distro main" > /etc/apt/sources.list.d/z-wave-me.list
+RUN wget -q -O - https://raw.githubusercontent.com/msazanov/Z-WayInstallScript/main/RaspbianInstall |bash
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile:13]: Wrong path. Should be https://storage.z-wave.me/Z-Way-Install

File: zwave_me_server/Dockerfile [hunk 2]
```diff
@@ -6,21 +6,21 @@ ENV distro=focal
 WORKDIR /opt/z-way-server
 
 RUN apt-get update && \
-    apt-get install -y wget procps gpg libcurl4 logrotate tzdata
+    apt-get install -qqy --no-install-recommends \
+    ca-certificates curl libnss-mdns \
+    wget procps gpg iproute2 openssh-client logrotate
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile:11]: _iproute2 openssh-client logrotate libnss-mdns_ are already in dependencies of zbw and z-way-server. please remove

File: zwave_me_server/Dockerfile [hunk 3]
```diff
@@ -6,21 +6,21 @@ ENV distro=focal
...
 
-# Add Z-Wave.Me repository
-RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 5b2f88a91611e683 && \
-    echo "deb https://repo.z-wave.me/z-way/$branch $distro main" > /etc/apt/sources.list.d/z-wave-me.list
+RUN wget -q -O - https://raw.githubusercontent.com/msazanov/Z-WayInstallScript/main/RaspbianInstall |bash
+RUN rm -f /opt/z-way-server/automation/storage/*
```
Comment (PoltoS) [inline zwave_me_server/Dockerfile:14]: Why do we need this?

File: zwave_me_server/config.yaml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 name: Z-Wave.Me Add-on
 startup: services
-version: 1.1.7
+version: 1.1.8
 slug: zwave_me_server
 description: 'Z-Wave.Me Add-on for HAss'
```
Comment (PoltoS) [inline zwave_me_server/config.yaml:5]: description: 'Z-Wave.Me Z-Way add-on for Home Assistant'

File: zwave_me_server/config.yaml [hunk 5]
```diff
@@ -13,10 +13,12 @@ ports:
...
+  configjson_device_replace: false
 schema:
   device: device(subsystem=tty)
+  configjson_device_replace: bool?
 webui: http://[HOST]:[PORT:8083]
 ports_description:
   8083/tcp: Z-Way UI
   8084/tcp: Webif UI
-image: "ghcr.io/z-wave-me/{arch}-ha-z-wave-me-addon"
+image: "ghcr.io/msazanov/{arch}-ha-z-wave-me-addon"
```
Comment (PoltoS) [inline zwave_me_server/config.yaml:24]: should be removed before making a PR

File: zwave_me_server/run.sh [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  # Get device path from Configuration tab of addon
+  device=$(grep -Eo '/dev/tty[A-Z]*[0-9]' /data/options.json)
+  forceDevice=$(grep "configjson_device_replace" /data/options.json | cut -d ":" -f 2)
+fi
+
+# Change device path in /defailtConfig/config.json
+sed -Ei "s|/dev/tty[A-Z]*[0-9]|$device|g" $defCJ
+
+# Folders and files that we need to copy to the external /data directory
+paths="/opt/z-way-server/automation/storage /opt/z-way-server/automation/userModules /etc/zbw /etc/init.d/zbw_connect /opt/z-way-server/automation/.syscommands"
```
Comment (PoltoS) [inline zwave_me_server/run.sh:24]: add skins and icons folders

File: zwave_me_server/run.sh [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+# Creating symlinks to the objects that we moved
+  ln -sf /data$path $path
+done
+
+# User configjson file path
+userCJ=$(ls /opt/z-way-server/automation/storage/configjson-*)
+
+# If we can't find configjson file in /data dirrectory we think this is first start and generate new remote ID
+if [ ! -f /data$userCJ ]; then 
+  wget 'http://find.z-wave.me/zbw_new_user?box_type=razberry' -O /tmp/zbw_connect_setup.run
```
Comment (PoltoS) [inline zwave_me_server/run.sh:44]: This is done by zbw_autosetup

File: zwave_me_server/run.sh [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+else
+  # Jut remove config folder from docker image dirrectory
+  rm -rf /opt/z-way-server/config/
+fi
+# Make simlinks
+ln -sf /data/opt/z-way-server/configs/ /opt/z-way-server/
+
+# Starting services
+/etc/init.d/dbus start
+#/etc/init.d/avahi-daemon start
```
Comment (PoltoS) [inline zwave_me_server/run.sh:70]: In that case also comment out dbus. I would suggest to uncomment both

**Review by PoltoS — CHANGES_REQUESTED** [review]

File: zwave_me_server/run.sh [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  wget 'http://find.z-wave.me/zbw_new_user?box_type=razberry' -O /tmp/zbw_connect_setup.run 
+  bash /tmp/zbw_connect_setup.run -y
+fi
+
+# If force device selection checked and /storage/configjson-* exists, replace the path in it too
+if [ -f /opt/z-way-server/automation/storage/configjson-* ] && [ $forceDevice ]; then
+  sed -Ei "s|/dev/tty[A-Z]*[0-9]|$device|g" $userCJ
+fi
+
+# Change default config dirrectory to configs/config for compapility in HA and HUB
```
Comment (PoltoS) [inline zwave_me_server/run.sh:53]:
```suggestion
# Change default config directory to configs/config for compatility with Docker, HA and Hub
```

File: zwave_me_server/config.yaml (hunk 4 above)
Comment (PoltoS) [inline zwave_me_server/config.yaml:5]:
```suggestion
description: 'Z-Wave.Me Z-Way add-on for Home Assistant'
```

File: zwave_me_server/Dockerfile (hunk 1 above)
Comment (PoltoS) [inline zwave_me_server/Dockerfile:13]:
```suggestion
RUN wget -q -O - https://storage.z-wave.me/Z-Way-Install |bash
```

File: zwave_me_server/config.yaml (hunk 5 above)
Comment (PoltoS) [inline zwave_me_server/config.yaml:24]: should be reverted to z-wave-me before making a PR

```suggestion
image: "ghcr.io/z-wave-me/{arch}-ha-z-wave-me-addon"
```

**Review by PoltoS — CHANGES_REQUESTED** [review]

File: zwave_me_server/run.sh (hunk 7 above)
Comment (msazanov) [reply zwave_me_server/run.sh:44]: As far as I remember, we used this to ensure that each running image had a different ID

**Review by PoltoS — APPROVED** [review]

**Pull request merged**